src/proof/alethe/alethe_post_processor.cpp

```diff
@@ -60,6 +60,335 @@ bool AletheProofPostprocessCallback::update(Node res,
+        success &=
+            addAletheStep(AletheRule::AND_POS, vp2_i, vp2_i, {}, {}, *cdp);
+        premisesVP2.push_back(vp2_i);
+        notAnd.push_back(andNode.notNode());  // cl F (not (and F1 ... Fn))^i
+      }
+
+      Node vp2a = nm->mkNode(kind::SEXPR, notAnd);
+      success &= addAletheStep(
+          AletheRule::RESOLUTION, vp2a, vp2a, premisesVP2, {}, *cdp);
+
+      notAnd.erase(notAnd.begin() + 1);  //(cl (not (and F1 ... Fn))^n)
+      notAnd.push_back(children[0]);     //(cl (not (and F1 ... Fn))^n F)
+      Node vp2b = nm->mkNode(kind::SEXPR, notAnd);
+      success &=
+          addAletheStep(AletheRule::REORDER, vp2b, vp2b, {vp2a}, {}, *cdp);
+
+      Node vp3 = nm->mkNode(kind::SEXPR, d_cl, andNode.notNode(), children[0]);
+      success &= addAletheStep(
+          AletheRule::DUPLICATED_LITERALS, vp3, vp3, {vp2b}, {}, *cdp);
+
+      Node vp8 = nm->mkNode(
+          kind::SEXPR, d_cl, nm->mkNode(kind::IMPLIES, andNode, children[0]));
+
+      Node vp4 = nm->mkNode(kind::SEXPR, d_cl, vp8[1], andNode);
+      success &=
+          addAletheStep(AletheRule::IMPLIES_NEG1, vp4, vp4, {}, {}, *cdp);
+
+      Node vp5 = nm->mkNode(kind::SEXPR, d_cl, vp8[1], children[0]);
+      success &=
+          addAletheStep(AletheRule::RESOLUTION, vp5, vp5, {vp4, vp3}, {}, *cdp);
+
+      Node vp6 = nm->mkNode(kind::SEXPR, d_cl, vp8[1], children[0].notNode());
+      success &=
+          addAletheStep(AletheRule::IMPLIES_NEG2, vp6, vp6, {}, {}, *cdp);
+
+      Node vp7 = nm->mkNode(kind::SEXPR, d_cl, vp8[1], vp8[1]);
+      success &=
+          addAletheStep(AletheRule::RESOLUTION, vp7, vp7, {vp5, vp6}, {}, *cdp);
+
+      if (children[0] != nm->mkConst(false))
+      {
+        success &= addAletheStep(
+            AletheRule::DUPLICATED_LITERALS, res, vp8, {vp7}, {}, *cdp);
+      }
+      else
+      {
+        success &= addAletheStep(
+            AletheRule::DUPLICATED_LITERALS, vp8, vp8, {vp7}, {}, *cdp);
+
+        Node vp9 =
+            nm->mkNode(kind::SEXPR,
+                       d_cl,
+                       nm->mkNode(kind::EQUAL, vp8[1], andNode.notNode()));
+
+        success &=
+            addAletheStep(AletheRule::IMPLIES_SIMPLIFY, vp9, vp9, {}, {}, *cdp);
+
+        Node vp10 =
+            nm->mkNode(kind::SEXPR, d_cl, vp8[1].notNode(), andNode.notNode());
+        success &=
+            addAletheStep(AletheRule::EQUIV1, vp10, vp10, {vp9}, {}, *cdp);
+
+        success &= addAletheStep(AletheRule::RESOLUTION,
+                                 res,
+                                 nm->mkNode(kind::SEXPR, d_cl, res),
+                                 {vp8, vp10},
+                                 {},
+                                 *cdp);
+      }
+
+      return success;
+    }
+    // The rule is translated according to the theory id tid and the outermost
+    // connective of the conclusion F. This is not an exact translation but
+    // should work in most cases.
```

Comment:
```suggestion
    // The rule is translated according to the theory id tid and the outermost
    // connective of the first term in the conclusion F, since F always has the
    // form (= t1 t2) where t1 is the term being rewritten. This is not an exact
    // translation but should work in most cases.
```